packages/harmony/src/components/popup/types.ts

```diff
@@ -87,6 +94,11 @@ export type PopupProps = {
    */
   checkIfClickInside?: (target: EventTarget) => boolean
 
+  /**
+   * Status prop to trigger rerenders on status updates for loading and error states
+   */
+  status?: Status
+
```

Comment:
yea i agree. I needed the popup to rerender when the artist data was finished being fetched so that the position would be recalculated. this was a simple way to do it based on the status of the request that was changing the content, but it's definitely weird to be on the Popup component. Open to other options here to achieve the same effect

Reply:
I think instead of this, you could have a `isLoading` or `hasLoaded` bool?

Reply:
Updated Popup to automatically reposition based on size change so no prop needed 👍